EU4toV2/Source/V2World/Country/Country.cpp

```diff
@@ -0,0 +1,1033 @@
+{
+	if (!details.civilized) return;
+
+	techs.insert("classicism_n_early_romanticism");
+	inventions.insert("carlism");
+	techs.insert("late_enlightenment_philosophy");
+	inventions.insert("declaration_of_the_rights_of_man");
+	if (normalizedScore >= -0.333)
+	{
+		techs.insert("enlightenment_thought");
+		inventions.insert("paternalism");
+		inventions.insert("constitutionalism");
+		inventions.insert("atheism");
+		inventions.insert("egalitarianism");
+		inventions.insert("rationalism");
+		inventions.insert("caste_privileges");
+		inventions.insert("sati_abolished");
+		inventions.insert("pig_fat_cartridges");
+	}
+	if (normalizedScore >= 0.333)
+	{
+		techs.insert("malthusian_thought");
+	}
+	if (normalizedScore >= 0.333)
+	{
+		techs.insert("introspectionism");
+	}
+	if (normalizedScore >= 0.666)
+	{
+		techs.insert("romanticism");
+		inventions.insert("romanticist_literature");
+		inventions.insert("romanticist_art");
+		inventions.insert("romanticist_music");
+	}
+}
+
+bool V2::Country::addFactory(std::shared_ptr<Factory> factory)
+{
+	// check factory techs
+	std::string requiredTech = factory->getRequiredTech();
+	if (!requiredTech.empty())
+	{
+		if (!techs.count(requiredTech)) return false;
+	}
+
+	// check factory inventions
+	std::string requiredInvention = factory->getRequiredInvention();
+	if (!requiredInvention.empty())
+	{
+		if (!inventions.count(requiredInvention)) return false;
+	}
+
+	// find a state to add the factory to, which meets the factory's requirements
+	std::vector<std::pair<double, std::shared_ptr<State>>> candidates;
+	for (const auto& candidate: states)
+	{
+		if (candidate->isColonial()) continue;
+		if (candidate->getFactoryCount() >= 8) continue;
+		if (factory->requiresCoastal()) if (!candidate->isCoastal()) continue;
+		if (!candidate->hasLandConnection()) continue;
+
+		double candidateScore = candidate->getSuppliedInputs(factory) * 100;
+		candidateScore -= static_cast<double>(candidate->getFactoryCount()) * 10;
+		candidateScore += candidate->getMfgRatio();
+		candidates.emplace_back(std::pair<double, std::shared_ptr<State>>(candidateScore, candidate));
+	}
+
+	sort(candidates.begin(), candidates.end(), factoryCandidateSortPredicate);
+
+	if (candidates.empty()) return false;
+
+	candidates[0].second->addFactory(factory);
+	details.numFactories++;
+	return true;
+}
+
+bool V2::Country::factoryCandidateSortPredicate(const std::pair<double, std::shared_ptr<State>>& lhs, const std::pair<double, std::shared_ptr<State>>& rhs)
+{
+	if (lhs.first != rhs.first) return lhs.first > rhs.first;
+	return lhs.second->getID() < rhs.second->getID();
+}
+
+void V2::Country::setupPops(
+	double popWeightRatio,
+	int popConversionAlgorithm,
+	const mappers::ProvinceMapper& provinceMapper
+) {
+	// skip entirely for empty nations
+	if (states.empty()) return;
+
+	// create the pops
+	for (const auto& province: provinces) province.second->doCreatePops(popWeightRatio, this, popConversionAlgorithm, provinceMapper);
+}
```

Comment:
Might be worth shoving the commented-out code for pop stats into a function and calling it conditional on theConfiguration.getDebug(). It'll be useful if ever reconsidering pop balance.

Or you might delete it and recreate it if ever needed, too.

Reply:
The commented out code went invalid a long time ago I fear. If we need it again we'll recreate it.